### PR TITLE
feat: add support for other markdown file extensions

### LINKED ISSUE #5148: `*.markdown` doesn't seem to be supported.
### What version of `astro` are you using?

1.5.2

### Are you using an SSR adapter? If so, which one?

None

### What package manager are you using?

NPM

### What operating system are you using?

MAC OSX 12.3.1

### Describe the Bug

I'm trying to rebuild my site using Astro and my site includes a blog. When building it today, I tried to use a `*.markdown` extension on my blog posts which I copied over from my old Jekyll blog. Astro threw an error saying that the content of the Markdown files wasn't `valid JS syntax`. I used the basic blog template.

### Link to Minimal Reproducible Example

https://github.com/angeldollface/angeldollface.github.io

### Participation

- [ ] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by MoustaphaDev**
## Changes

Closes #5148. Adds support for the following markdown file extensions:
- `.markdown`
- `.mdown`
- `.mkdn`
- `.mkd`
- `.mdwn`

## Testing

- The `markdown.test.js` test suite was updated to include test coverage for markdown pages with the new extensions.
- A test case in `rss.test.js` was updated to match the updated error message.

## Docs

[withastro/docs#1893](https://github.com/withastro/docs/pull/1893)

### CODE REVIEW

File: packages/astro/test/astro-markdown.test.js [hunk 1]
```diff
@@ -14,6 +14,12 @@ describe('Astro Markdown', () => {
 		await fixture.build();
 	});
 
+	it('Can process `.markdown` files', async () => {
+		const html = await fixture.readFile('/page-with-markdown-extension/index.html');
+		const $ = cheerio.load(html);
+		expect($('h1').html()).to.equal('Page with markdown extension');
+	});
+
```
Comment (MoustaphaDev) [inline packages/astro/test/astro-markdown.test.js:22]: Maybe we need more test cases?

File: packages/astro/test/markdown.test.js [hunk 2]
```diff
@@ -17,6 +17,13 @@ describe('Markdown tests', () => {
 			await fixture.build();
 		});
 
+		it('Can load a `.markdown` file', async () => {
+			const html = await fixture.readFile('/page-with-markdown-extension/index.html');
+			const $ = cheerio.load(html);
+			expect($('h1').html()).to.equal('Page with markdown extension');
+			expect($('p').html()).to.equal('Hope this loads fine 🤞');
+		});
```
Comment (MoustaphaDev) [inline packages/astro/test/markdown.test.js:25]: Maybe we need more test cases?

**Comment by MoustaphaDev** [comment]
I feel like the mdx and rss integrations don't need a changeset, tell me I'll revert that

**Comment by mrienstra** [comment]
One more small change needed:

[packages/astro-rss/test/rss.test.js#L201](https://github.com/withastro/astro/blob/6e1d62f/packages/astro-rss/test/rss.test.js#L201)
```diff
-				chai.expect(err.message).to.contain('you can only glob ".md" files within /pages');
+				chai.expect(err.message).to.contain('you can only glob ".md" or ".markdown" files within /pages');
```

**Comment by sarah11918** [comment]
@MoustaphaDev - This looks great from the perspective of Docs-facing stuff in files for this PR!

As for what needs to go directly in Docs, I believe it should just be the `markdown-content` page that needs to be updated, and only where we "define" Markdown files as `.md` files. There are some examples that use an `.md` file, which can stay as is. That's the file being used as the example, and we're describing it. But, there are a handful of places where we say "Markdown (`.md`)..." where we should absolutely tack on `.markdown` as another possible option. 

I even checked the Layouts page which has a Markdown Layouts section, and we only say "Markdown" (we don't mention the file extension), so I think just targeting `markdown-content` in the docs makes sense.

Are you willing to make a Docs PR, or do you want someone from Team Docs to take this on?

**Comment by MoustaphaDev** [comment]
Thanks @sarah11918  for pointing me to the right direction, I'll open a docs PR to include this update.

**Comment by Princesseuh** [comment]
There's a few more extensions that are sometimes used for Markdown. GitHub for instance supports the following: `.markdown, .mdown, .mkdn, .mkd, .mdwn, .md`. I'm not sure how common they are, but maybe it'd be worth it to add them even then to avoid further issues in the future?

**Comment by MoustaphaDev** [comment]
> There's a few more extensions that are sometimes used for Markdown. GitHub for instance supports the following: `.markdown, .mdown, .mkdn, .mkd, .mdwn, .md`. I'm not sure how common they are, but maybe it'd be worth it to add them even then to avoid further issues in the future?

Right, better support all the options, I'll make some changes in a bit

File: .changeset/spicy-cameras-rush.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
+'astro': patch
```
Comment (matthewp) [inline .changeset/spicy-cameras-rush.md:2]: can you make this be `minor` instead?

File: .changeset/spicy-cameras-rush.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+'@astrojs/rss': patch
+'@astrojs/mdx': patch
+---
+
+Support `.markdown` file extension for markdown files
```
Comment (matthewp) [inline .changeset/spicy-cameras-rush.md:7]: Since you changed this to support other extensions, how about a list?

**Comment by matthewp** [comment]
Code looks very clean! Think this should be marked as `minor` though, which would put it at a Thursday release if ready by EOD Wednesday.

File: packages/astro-rss/src/index.ts [hunk 5]
```diff
@@ -58,7 +58,7 @@ function mapGlobResult(items: GlobResult): Promise<RSSFeedItem[]> {
 			const { url, frontmatter } = await getInfo();
 			if (url === undefined || url === null) {
 				throw new Error(
-					`[RSS] When passing an import.meta.glob result directly, you can only glob ".md" files within /pages! Consider mapping the result to an array of RSSFeedItems. See the RSS docs for usage examples: https://docs.astro.build/en/guides/rss/#2-list-of-rss-feed-objects`
+					`[RSS] When passing an import.meta.glob result directly, you can only glob ".md" (or alternative extensions for markdown files like ".markdown") files within /pages! Consider mapping the result to an array of RSSFeedItems. See the RSS docs for usage examples: https://docs.astro.build/en/guides/rss/#2-list-of-rss-feed-objects`
```
Comment (MoustaphaDev) [inline packages/astro-rss/src/index.ts:61]: Should cause a patch for the rss package, right?

Comment (matthewp) [reply packages/astro-rss/src/index.ts:61]: Yeah, I think that's file.

File: packages/astro/test/markdown.test.js (hunk 2 above)
Comment (matthewp) [reply packages/astro/test/markdown.test.js:25]: I think this is ok to start, i'm not super worried about this change breaking

**Comment by matthewp** [comment]
@MoustaphaDev can you remove the `Draft` when you think this is ready? Thanks!

**Comment by MoustaphaDev** [comment]
> @MoustaphaDev can you remove the `Draft` when you think this is ready? Thanks!

Will do in a moment, I'll just add some test cases for the new extensions.

**Review by delucis — APPROVED** [review]
Looks good from a docs perspective!

**Pull request merged**